source/functions/test.txt

```diff
@@ -0,0 +1,313 @@
+        // your mocking services
+      };
+    });
+
+    // removes context from global namespace after each test
+    afterEach(() => {
+      delete global.context;
+    });
+
+    test("should perform operation using App Services globals", () => {
+      // test function that uses context
+    });
+
+.. example:: Mocking a Function that accesses ``context``
+
+   The function in this example accesses an App Services Value and returns it.
+
+   .. literalinclude:: /examples/generated/functions/helloWithValue.snippet.hello-with-context-value.js
+      :language: js
+      :caption: helloWithValue.js
+      :emphasize-lines: 2
+
+   Now create a test file ``helloWithValue.test.js``.
+   The test file contains the following:
+
+   - Import the function exported from ``helloWithValue.js``.
+   - A mock of ``context.values.get()``. Wrap the mock in set up and tear down blocks
+     so that it does not pollute the global namespace.
+   - A test of the imported function that uses the mock.
+
+   .. literalinclude:: /examples/generated/functions/helloWithValue.test.snippet.test-with-mocking.js
+      :language: js
+      :caption: helloWithValue.test.js
+
+.. _integration-test-functions:
+
+Integration Tests for Functions
+-------------------------------
+
+You should perform integration tests on all Functions before deploying
+them to production environments. This is especially important because
+the Atlas Function JavaScript runtime differs from the standard Node.js runtime.
+Unexpected errors can occur if you do not test functions deployed to App Services.
+
+There is no single way to write integration tests for Functions.
+As Functions can be used in a variety of different contexts for different purposes,
+each use case requires a different integration test strategy.
+
+For example, the way you create an integration test for a Function that you
+invoke from a Realm client SDK is different from the way you test
+a GraphQL Custom Resolver Function, which is different from how you would
+test a Database Trigger Function.
+
+However, there are some general steps that you can take to writing integration tests
+for Functions. On a high level these steps are:
+
+#. Create a testing App Services App with the same configuration as your production
+   App Services App.
+#. Write integration tests that use the Realm client SDKs to interact with the test App.
+
+The remainder of this section explains how to implement integration tests for your
+App in more detail.
+
+.. seealso::
+
+   For more information on the unique aspects of the Functions JavaScript runtime,
+   refer to:
+
+   - :ref:`Function Constraints <function-constraints>`
+   - :ref:`JavaScript Support <js-support>`
+
+   For more information on the different use cases for Functions,
+   refer to :ref:`When to Use Functions <when-to-use-functions>`.
+
+.. procedure::
+
+   .. step:: Create a test App
+
+      Create an App Services App for testing purposes that has the same configuration
```

Comment:
"same configuration" here could be misleading -- I wonder if we should explain that you'll need to use a test data source and test versions of any other backend stuff (auth, for instance) as well.